src/Web/WebStatus/Program.cs

```diff
@@ -70,10 +71,7 @@ IConfiguration GetConfiguration()
 
     if (config.GetValue<bool>("UseVault", false))
     {
-        builder.AddAzureKeyVault(
-            $"https://{config["Vault:Name"]}.vault.azure.net/",
-            config["Vault:ClientId"],
-            config["Vault:ClientSecret"]);
+        builder.AddAzureKeyVault(new Uri($"https://{config["Vault:Name"]}.vault.azure.net/"), new DefaultAzureCredential());
```

Comment:
This change is not equivalent. The `DefaultAzureCredential` would work in the environments that have managed identity or were authenticated via some other mechanism (Az PowerShell, az cli, VS etc.) but it won't pick up `Vault:ClientId`/`Vault:ClientSecret` values from the config automatically.

Reply:
Thanks for your review! Could you review the new method I used in the code?